File: postgresql/golang/repositories/postgres_repository.go
package repositories

import (
	"context"

	modelentities "note-golang-postgresql/models/entities"

	"github.com/jmoiron/sqlx"
)

type PostgresRepository interface {
	Create(tx *sqlx.Tx, ctx context.Context, test1 modelentities.Test1) (id int, err error)
	Get(db *sqlx.DB, ctx context.Context, id int) (test1 modelentities.Test1, err error)
	Update(tx *sqlx.Tx, ctx context.Context, test1 modelentities.Test1) (rowsAffected int64, err error)
	Delete(tx *sqlx.Tx, ctx context.Context, id int) (rowsAffected int64, err error)
}

type postgresRepository struct {
}

func NewPostgresRepository() PostgresRepository {
	return &postgresRepository{}
}

func (repository *postgresRepository) Create(tx *sqlx.Tx, ctx context.Context, test1 modelentities.Test1) (id int, err error) {
	err = tx.GetContext(ctx, &id, `INSERT INTO test1 (test) VALUES ($1) RETURNING id;`, test1.Test)
	return
}

func (repository *postgresRepository) Get(db *sqlx.DB, ctx context.Context, id int) (test1 modelentities.Test1, err error) {
	err = db.GetContext(ctx, &test1, `SELECT id, test FROM test1 WHERE id = $1;`, id)
	return
}

func (repository *postgresRepository) Update(tx *sqlx.Tx, ctx context.Context, test1 modelentities.Test1) (rowsAffected int64, err error) {
	result, err := tx.ExecContext(ctx, `UPDATE test1 SET test = $1 WHERE id = $2;`, test1.Test, test1.Id)
	if err != nil {
		return
	}
	return result.RowsAffected()
}

func (repository *postgresRepository) Delete(tx *sqlx.Tx, ctx context.Context, id int) (rowsAffected int64, err error) {
	result, err := tx.ExecContext(ctx, `DELETE FROM test1 WHERE id = $1;`, id)
	if err != nil {
		return
	}
	return result.RowsAffected()
}

File: postgresql/rust/src/repositories/postgres_repository.rs
use sqlx::{Postgres, Transaction};

use crate::models::entities::test1::Test1;

pub trait PostgresRepository {
    async fn create(&self, tx: &mut Transaction<'_, Postgres>, test1: &Test1) -> Result<i32, sqlx::Error>;
    async fn get_by_id(&self, pool: &sqlx::PgPool, id: i32) -> Result<Test1, sqlx::Error>;
    async fn update(&self, tx: &mut Transaction<'_, Postgres>, test: &Test1) -> Result<u64, sqlx::Error>;
    async fn delete(&self, tx: &mut Transaction<'_, Postgres>, id: i32) -> Result<u64, sqlx::Error>;
}

pub struct PostgresRepositoryImpl{}

impl PostgresRepositoryImpl {
    pub fn new() -> PostgresRepositoryImpl {
        PostgresRepositoryImpl {  }
    }
}

impl PostgresRepository for PostgresRepositoryImpl {
    async fn create(&self, tx: &mut Transaction<'_, Postgres>, test1: &Test1) -> Result<i32, sqlx::Error> {
        let result: (i32,) = sqlx::query_as("INSERT INTO test1(test) VALUES($1) RETURNING id;")
            .bind(&test1.test)
            .fetch_one(&mut **tx)
            .await?;
        Ok(result.0)
    }

    async fn get_by_id(&self, pool: &sqlx::PgPool, id: i32) -> Result<Test1, sqlx::Error> {
        let test1 = sqlx::query_as::<_, Test1>("SELECT id, test FROM test1 WHERE id = $1;")
            .bind(id)
            .fetch_one(pool)
            .await?;
        Ok(test1)
    }

    async fn update(&self, tx: &mut Transaction<'_, Postgres>, test1: &Test1) -> Result<u64, sqlx::Error> {
        let result = sqlx::query("UPDATE test1 SET test = $1 WHERE id = $2;")
            .bind(&test1.test)
            .bind(&test1.id)
            .execute(&mut **tx)
            .await?;
        Ok(result.rows_affected())
    }

    async fn delete(&self, tx: &mut Transaction<'_, Postgres>, id: i32) -> Result<u64, sqlx::Error> {
        let result = sqlx::query("DELETE FROM test1 WHERE id = $1;")
            .bind(id)
            .execute(&mut **tx)
            .await?;
        Ok(result.rows_affected())
    }
}